spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java

```diff
@@ -113,17 +119,12 @@ public RootController rootController(EntityLinks entityLinks) {
 	}
 
 	@Bean
-	public RuntimeAppsController runtimeAppsController(StreamDefinitionRepository repository,
-			DeploymentIdRepository deploymentIdRepository, AppDeployer appDeployer) {
-		return new RuntimeAppsController(repository, deploymentIdRepository, appDeployer);
-	}
-
-	@Bean
 	public AppInstanceController appInstanceController(AppDeployer appDeployer) {
 		return new AppInstanceController(appDeployer);
 	}
 
 	@Bean
+	@ConditionalOnBean(StreamDefinitionRepository.class)
```

Comment:
Currently, this configuration class acts as `AutoConfiguration` class to instantiate all the REST controllers in a single place. Moving these into separate configuration class might still work but it might be too confusing at this point I think.